### ISSUE TITLE
Test Adapter for Test Explorer throws System.AggregateException

### ISSUE THREAD

**Issue by SeanSobey** [mmanela/chutzpah#460]
Hi

Since VS 2015 tests that ran fine in the VS 2013 adapter are now throwing errors. They work fine on our build machine through TFS running Chutzpah so I assume it is just the adapter that has issues.

Some investigation indicates that a few scenarios cause the errors but they all give an Interop COM message:
<code>Error: System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: The specified test case was not found in the current result context (mapping test id to test node).</code>

This is the full output from the Tests window:
```
------ Run test started ------
Error: System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Chutzpah.TestCaseStreamReader.Read(ProcessStream processStream, TestOptions testOptions, TestContext testContext, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestRunner.<>c__DisplayClass1d.<InvokeTestRunner>b__1a(ProcessStream processStream)
   at Chutzpah.ProcessHelper.RunExecutableAndProcessOutput[T](String exePath, String arguments, Func`2 streamProcessor)
   at Chutzpah.TestRunner.InvokeTestRunner(String headlessBrowserPath, TestOptions options, TestContext testContext, TestExecutionMode testExecutionMode, ITestMethodRunnerCallback callback)
   at Chutzpah.TestRunner.<>c__DisplayClassc.<ExecuteTestContexts>b__b(TestContext testContext)
---> (Inner Exception #0) System.Runtime.InteropServices.COMException (0x80070490): The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()<---

While Running:c:\tfs\sirc\core\sage.intelligence.officeaddin.tests\src\test.tests.js

========== Run test finished: 0 run (0:00:03.8244173) ==========
```

This can be caused by running the tests from the Test Explorer.

Interestingly it appears a single test runs ok. This can be replicated in a new project with a simple test file (example.tests.js):

```
(function () {
	"use strict";
	
	describe("Tests", function () {

		it("tests A", function () {

			expect(false).toEqual(true);
		});
		it("tests B", function () {

			expect(false).toEqual(true);
		});
	});
})();
```

I have a chutzpah.json as well if its relevant:

```
{
	"RootReferencePathMode": "SettingsFileDirectory",
	"Tests": [
		{
			"Path": "src",
			"Includes": [ "*.tests.js" ],
			"ExpandReferenceComments": true
		}
	]
}
```

**Comment by mmanela** [comment]
Thanks for reporting the issue. I tried to follow your steps and repro but I could not. I was able to run the tests you posted and get the two failures fine. Here is the code I used:

https://github.com/mmanela/Issue-Chutzpah-460

Let me know if I am missing something.

**Comment by SeanSobey** [comment]
Thanks for taking a look.

On that same repo I get the same Issue:
<code>
------ Run test started ------
Error: System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Chutzpah.TestCaseStreamReader.Read(ProcessStream processStream, TestOptions testOptions, TestContext testContext, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestRunner.<>c__DisplayClass1d.<InvokeTestRunner>b__1a(ProcessStream processStream)
   at Chutzpah.ProcessHelper.RunExecutableAndProcessOutput[T](String exePath, String arguments, Func`2 streamProcessor)
   at Chutzpah.TestRunner.InvokeTestRunner(String headlessBrowserPath, TestOptions options, TestContext testContext, TestExecutionMode testExecutionMode, ITestMethodRunnerCallback callback)
   at Chutzpah.TestRunner.<>c__DisplayClassc.<ExecuteTestContexts>b__b(TestContext testContext)
---> (Inner Exception #0) System.Runtime.InteropServices.COMException (0x80070490): The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()<---

While Running:c:\users\sean.sobey\documents\github\issue-chutzpah-460\src\example.tests.js

========== Run test finished: 1 run (0:00:03.6805413) ==========
</code>

I am using VS 2015 Update 1, test Adapter v4.1.0. Is there any other debug stuff I can try?

**Comment by mmanela** [comment]
I am also running VS 2015 update 1 and  Test Adapter 4.1.0.

1. Do you have any other extensions installed? Please list them

2. Can you turn chutzpah tracing on and share the log file. In Tools->Options->Chutzpah->Unit Test Explorer there is a tracing options. This will trace to %temp%\chutzpah.log. Make sure you delete this file before your test to make sure it doesn't include results from other runs.

3. Are you able to run the tests using the other chutzpah plugin? The one that just adds the context menu.

4. Did you try uninstalling Chutzpah, purging extension cache and re-installing?

**Comment by SeanSobey** [comment]
I think I have found the cause of the issue, while going through the steps you mentioned.

tl; dr: Its [Productivity Power Tools 2015 for Visual Studio 2015](https://visualstudiogallery.msdn.microsoft.com/34ebc6a2-2777-421d-8914-e29c1dfa7f5d).

1) Yep, loads, but I unloaded all (except the ones that cannot be disabled), and prepared a list of the remaining ones. But I found in this process that disabling the above-mentioned plugin(Productivity Power Tools 2015 for Visual Studio 2015) caused Chutzpah to work fine. Then re-enabling breaks it again.

2) I did this and created the log fine (it contained the same exception I posted), I can put this up if you still require it.

3) Yes the command line runner plugin works fine it looks like.

4) Did this before the original post and did again a few times now.

**Comment by mmanela** [comment]
Yes please do. Chutzpah has worked with this on the past so I wonder if they did something to break Chutzpah.

**Comment by SeanSobey** [comment]
Logged in there Q&A section on their [VS Extension site](https://visualstudiogallery.msdn.microsoft.com/34ebc6a2-2777-421d-8914-e29c1dfa7f5d/view/Discussions) (could not find any public repo).

Thanks for all your help!

**Comment by dazinator** [comment]
I can reproduce this exception, and I don't have the productivity power tools installed..

I have a `dnx` based project, and I get the exception as soon as I add additional tests to my ts file, and try to run those additional tests individually.

The first test runs individually, all other additional tests fail with this exception.

Interestingly I can run the whole group of tests fine though.

I.e my tests:

```
/// <reference path="../typings/qunit/qunit.d.ts" />
/// <reference path="../typings/requirejs/require.d.ts" />

import core = require("Core");

QUnit.module("Core");
test("will return correct version from core", function () {
    var version = core.version;
    equal(version, 8);
});


test("will fail", function () {
    var version = core.version;
    equal(version, 8);
});

```

Running the the whole group of tests works:

![image](https://cloud.githubusercontent.com/assets/3176632/13680377/353e0988-e6f0-11e5-9c75-189b5fc9f067.png)

Running the first test on its own works:

![image](https://cloud.githubusercontent.com/assets/3176632/13680394/4a2c75dc-e6f0-11e5-9afb-ea05e2e0fb57.png)

Running the second test on it's own fails:

```

------ Run test started ------
Error: System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Chutzpah.TestCaseStreamReader.Read(ProcessStream processStream, TestOptions testOptions, TestContext testContext, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestRunner.<>c__DisplayClass26.<InvokeTestRunner>b__23(ProcessStream processStream)
   at Chutzpah.ProcessHelper.RunExecutableAndProcessOutput[T](String exePath, String arguments, Func`2 streamProcessor)
   at Chutzpah.TestRunner.InvokeTestRunner(String headlessBrowserPath, TestOptions options, TestContext testContext, TestExecutionMode testExecutionMode, ITestMethodRunnerCallback callback)
   at Chutzpah.TestRunner.<>c__DisplayClass15.<ExecuteTestContexts>b__14(TestContext testContext)
---> (Inner Exception #0) System.Runtime.InteropServices.COMException (0x80070490): The specified test case was not found in the current result context (mapping test id to test node).
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
   at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
   at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
   at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
   at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()<---

```

It's nothing to do with the tests themselves - if I reverse their order, the exact same issue occurs, i.e the first test (previously the second test) can now run individually, but the second test (previously the first) will hit the issue.

Do you need a repro?

**Comment by mmanela** [comment]
This issue will be fixed in the next update of visual studio. (VS 2015 Update 2). I confirmed the fix is commited by the team that runs the test adapter.

**Comment by jairov4** [comment]
Hi, I am using Visual Studio Update 2, but the problem still there. Is something wrong inside vstest.console? I was trying to get this working in TFS too.

**Comment by mmanela** [comment]
Are you seeing the exact same issue on vs 2015 update 2? Can I see the error stack trace?

**Comment by jairov4** [comment]
Sure, this makes sense for you?

    Error: Error: System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: The test case was not recognized to be part of any test file.
       at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
       at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
       at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
       at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
       at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
       at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.Execute()
       --- End of inner exception stack trace ---
       at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
       at System.Threading.Tasks.Task`1.get_Result()
       at Chutzpah.TestCaseStreamReader.Read(ProcessStream processStream, TestOptions testOptions, TestContext testContext, ITestMethodRunnerCallback callback, Boolean debugEnabled)
       at Chutzpah.TestRunner.<>c__DisplayClass26.<InvokeTestRunner>b__23(ProcessStream processStream)
       at Chutzpah.ProcessHelper.RunExecutableAndProcessOutput[T](String exePath, String arguments, Func`2 streamProcessor)
       at Chutzpah.TestRunner.InvokeTestRunner(String headlessBrowserPath, TestOptions options, TestContext testContext, TestExecutionMode testExecutionMode, ITestMethodRunnerCallback callback)
       at Chutzpah.TestRunner.<>c__DisplayClass15.<ExecuteTestContexts>b__14(TestContext testContext)
    ---> (Inner Exception #0) System.Runtime.InteropServices.COMException (0x80070490): The test case was not recognized to be part of any test file.
       at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
       at WEX.TestExecution.FrameworkHandle.RecordStart(TestCase testCase)
       at Chutzpah.VS2012.TestAdapter.ExecutionCallback.TestStarted(TestCase test)
       at Chutzpah.Callbacks.ParallelRunnerCallbackAdapter.TestStarted(TestCase testCase)
       at Chutzpah.TestCaseStreamReader.ReadFromStream(StreamReader stream, TestContext testContext, TestOptions testOptions, IList`1 streamingTestFileContexts, IList`1 deferredEvents, ITestMethodRunnerCallback callback, Boolean debugEnabled)
       at Chutzpah.TestCaseStreamReader.<>c__DisplayClass8.<Read>b__2()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.Execute()<---

**Comment by mmanela** [comment]
That looks like a different error. Can you provide a repro I can run?
